skimage/segmentation/_clear_border.py

```diff
@@ -1,37 +1,39 @@
 import numpy as np
-from scipy.ndimage import label
+#from scipy.ndimage import label
+from ..measure import label
 
 
-def clear_border(image, buffer_size=0, bgval=0):
-    """Clear objects connected to image border.
+def clear_border(labels, buffer_size=0, bgval=0):
+    """Clear objects connected to the label image border.
 
-    The changes will be applied to the input image.
+    The changes will be applied directly to the input.
 
     Parameters
     ----------
-    image : (N, M) array
-        Binary image.
+    labels : (N, M) array of int
+        Label or binary image.
     buffer_size : int, optional
-        Define additional buffer around image border.
+        The width of the border examined.  By default, only objects
+        that touch the outside of the image are removed.
     bgval : float or int, optional
-        Value for cleared objects.
+        Cleared objects are set to this value.
 
     Returns
     -------
-    image : (N, M) array
-        Cleared binary image.
+    labels : (N, M) array
+        Cleared binary image.  Note that the input label image is modified.
```

Comment:
@jni sorry that was not clear. I should have said +1 for having an ``in_place`` keyword argument, and it might indeed be better to have the default value set to False, to keep the "functional approach".